src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1564,18 +1558,20 @@ public sealed override async Task<HttpResponseMessage> SendAsync(HttpRequestMess
         {
             if (NetEventSource.IsEnabled) Trace($"{request}");
 
-            Http2Stream http2Stream = null;
             try
             {
                 // Send request headers
                 bool shouldExpectContinue = request.Content != null && request.HasHeaders && request.Headers.ExpectContinue == true;
-                http2Stream = await SendHeadersAsync(request, cancellationToken, mustFlush: shouldExpectContinue).ConfigureAwait(false);
+                Http2Stream http2Stream = await SendHeadersAsync(request, cancellationToken, mustFlush: shouldExpectContinue).ConfigureAwait(false);
+
+                bool duplex = request.Content != null && request.Content.AllowDuplex;
 
-                // Start sending the request body, if any.
-                Task requestBodyTask = 
-                    request.Content == null ? Task.CompletedTask :
-                    shouldExpectContinue ? http2Stream.SendRequestBodyWithExpect100ContinueAsync(cancellationToken) :
-                    http2Stream.SendRequestBodyAsync(cancellationToken);
+                // If we have duplex content, then don't propagate the cancellation to the request body task.
```

Comment:
Do we expect this to be true even once we add the relevant HttpContent.SerializeToStreamAsync overload that takes a CancellationToken?  If not, we should tag this with https://github.com/dotnet/corefx/issues/9071 so we remember to follow-up here when we do that.

Reply:
Even once we add the overload, I assume we will need to handle legacy HttpContent classes that don't override it.

Reply:
> Even once we add the overload, I assume we will need to handle legacy HttpContent classes that don't override it.

Sure.  But won't we want to do things differently for the case where it does?

Reply:
Differently how?

Reply:
This is explicitly not passing the token in. With the overload available, it could, regardless of whether the content overrides the method or not (if it doesn't, then the base impl will just delegate to the existing one, as it already does).

Reply:
Anyway, doesn't matter right now.